sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -84,6 +84,7 @@ def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
             raise ValueError("Please specify a table name.")
         _validate_table_name(table_name)
         self.table_name = table_name
+        self.prepare_key = _prepare_key
```

Comment:
Shouldn't the value be `None or _prepare_key`? Otherwise, why you are not using _prepare_key directly.

Reply:
By assigning the function to a public attribute on the client - a user can easily patch the behaviour to revert it, as is demonstrated in the tests:
```python
# This will replace the escaping function with a no-op, essentially reverting the fix.
client.prepare_key = lambda key: key
```